compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/ToolAnnotationAnalysisTask.java

```diff
@@ -0,0 +1,196 @@
+package io.ballerina.lib.np.compilerplugin;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ModuleSymbol;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.ExternalFunctionBodyNode;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.ReturnTypeDescriptorNode;
+import io.ballerina.openapi.service.mapper.type.TypeMapper;
+import io.ballerina.openapi.service.mapper.type.TypeMapperImpl;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.core.util.OpenAPISchema2JsonSchema;
+import io.swagger.v3.oas.models.media.Schema;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.lib.np.compilerplugin.Commons.findNPModule;
+import static io.ballerina.lib.np.compilerplugin.Commons.hasLlmCallAnnotation;
+import static io.ballerina.projects.util.ProjectConstants.EMPTY_STRING;
+
+public class ToolAnnotationAnalysisTask implements AnalysisTask<SyntaxNodeAnalysisContext> {
```

Comment:
Why do we need a separate analysis task for this? Why can't we do it in the modifier itself?

Reply:
First, we need to identify the types that require the annotation and then update their corresponding type definitions accordingly. Identification of types is part of the analysis process and does not related to the modification. Including this step within the modifier would complicate the logic. That is the main reason to add separate analyzer

Reply:
```
If the complication is doing it in the visitor, you can avoid that by collecting the types and doing the modification at the end.
```
Yes we can do this. But that is the same thing what I did in the analyzer.

Reply:
My point is that what we are doing as "analysis" is part of the modification logic. If the modification isn't happening, the analysis isn't required. So, I would merge the two.

Reply:
To obtain the JSON schema, we need to create an instance of TypeMapper(openapi), which requires passing a SyntaxNodeAnalysisContext. So we need to keep analyzer

Reply:
Updated as we discussed offline